### ISSUE TITLE
更新1.6.2版本后，php-fpm会时不时的出core

### ISSUE THREAD

**Issue by yyz26371945** [SeasX/SeasLog#100]
在php5.6.20环境下，更新1.6.2版本后，php-fpm会时不时的出core，大概10-30一次
[18-Jul-2016 19:36:19] WARNING: [pool www] child 26588 exited on signal 11 (SIGSEGV) after 859.294876 seconds from start
用gdb看了之后发现是seaslog产生的
Core was generated by `php-fpm: p'.
Program terminated with signal 11, Segmentation fault.
#0  0x00007fcce8f93bf1 in seaslog_init_buffer () at /home/batsdk/tools/software/SeasLog-SeasLog-1.6.2/seaslog.c:554

554         if (SEASLOG_G(buffer) && Z_TYPE_P(SEASLOG_G(buffer)) == IS_ARRAY)
Missing separate debuginfos, use: debuginfo-install cyrus-sasl-lib-2.1.23-13.el6.x86_64 glibc-2.12-1.80.el6.x86_64 keyutils-libs-1.4-4.el6.x86_64 krb5-libs-1.9-33.el6.x86_64 libcom_err-1.41.12-12.el6.x86_64 libcurl-7.19.7-26.el6_2.4.x86_64 libgcc-4.4.6-4.el6.x86_64 libidn-1.18-2.el6.x86_64 libpng-1.2.49-1.el6_2.x86_64 libselinux-2.0.94-5.3.el6.x86_64 libssh2-1.2.2-7.el6_2.3.x86_64 libstdc++-4.4.6-4.el6.x86_64 nspr-4.9-1.el6.x86_64 nss-3.13.3-6.el6.x86_64 nss-softokn-freebl-3.12.9-11.el6.x86_64 nss-util-3.13.3-2.el6.x86_64 openldap-2.4.23-26.el6.x86_64 openssl-1.0.0-20.el6_2.5.x86_64 zlib-1.2.3-27.el6.x86_64
(gdb) bt
#0  0x00007fcce8f93bf1 in seaslog_init_buffer () at /home/batsdk/tools/software/SeasLog-SeasLog-1.6.2/seaslog.c:554
#1  0x00007fcce8f95d13 in zm_activate_seaslog (type=<value optimized out>, module_number=<value optimized out>)

```
at /home/batsdk/tools/software/SeasLog-SeasLog-1.6.2/seaslog.c:265
```
#2  0x0000000000789330 in zend_activate_modules () at /home/batsdk/tools/software/php-5.6.20/Zend/zend_API.c:2414
#3  0x00000000007255ea in php_request_startup () at /home/batsdk/tools/software/php-5.6.20/main/main.c:1686
#4  0x000000000083351d in main (argc=<value optimized out>, argv=<value optimized out>) at /home/batsdk/tools/software/php-5.6.20/sapi/fpm/fpm/fpm_main.c:1927

**Comment by yyz26371945** [comment]
@Neeke 可否处理一下，不然该版本一直没办法使用，我看另外一个issue有内存泄露的问题~

**Comment by yyz26371945** [comment]
@Neeke 这个bug是不再修了？？？线上一直出core，只能用1.5.3版本的

**Comment by Neeke** [comment]
@yyz26371945   正在处理中。

**Comment by Neeke** [comment]
@yyz26371945  你能不能提供一个使用SeasLog时的例子，方便我重现问题。

**Comment by Neeke** [comment]
@yyz26371945   请使用Master版本测试一下，版本号为 1.6.4

**Comment by yyz26371945** [comment]
我这边没有具体的例子，每次core都是在不同的写日志位置，我感觉是并发较大的问题。我这边是大概600左右的qps

**Comment by yyz26371945** [comment]
是一台机器600左右，总共是6台，峰值能到4000

**Comment by Neeke** [comment]
请使用Master版本测试一下，版本号为 1.6.4

**Comment by yyz26371945** [comment]
还是不行，use_buffer等于0或者1都会产生错误，core信息如下：
Core was generated by `php-fpm: p'.
Program terminated with signal 11, Segmentation fault.
#0  0x00000000007a5ee9 in gc_zval_possible_root (zv=0x7fead04ce470) at /home/batsdk/software/tools/php-5.6.20/Zend/zend_gc.c:143
143                     GC_ZOBJ_CHECK_POSSIBLE_ROOT(zv);
Missing separate debuginfos, use: debuginfo-install cyrus-sasl-lib-2.1.23-13.el6.x86_64 glibc-2.12-1.80.el6.x86_64 keyutils-libs-1.4-4.el6.x86_64 krb5-libs-1.9-33.el6.x86_64 libcom_err-1.41.12-12.el6.x86_64 libcurl-7.19.7-26.el6_2.4.x86_64 libgcc-4.4.6-4.el6.x86_64 libidn-1.18-2.el6.x86_64 libpng-1.2.49-1.el6_2.x86_64 libselinux-2.0.94-5.3.el6.x86_64 libssh2-1.2.2-7.el6_2.3.x86_64 libstdc++-4.4.6-4.el6.x86_64 nspr-4.9-1.el6.x86_64 nss-3.13.3-6.el6.x86_64 nss-softokn-freebl-3.12.9-11.el6.x86_64 nss-util-3.13.3-2.el6.x86_64 openldap-2.4.23-26.el6.x86_64 openssl-1.0.0-20.el6_2.5.x86_64 zlib-1.2.3-27.el6.x86_64
(gdb) bt
#0  0x00000000007a5ee9 in gc_zval_possible_root (zv=0x7fead04ce470) at /home/batsdk/software/tools/php-5.6.20/Zend/zend_gc.c:143
#1  0x0000000000792c0b in zend_hash_destroy (ht=0x7feb285fb130) at /home/batsdk/software/tools/php-5.6.20/Zend/zend_hash.c:548
#2  0x0000000000784833 in _zval_dtor_func (zvalue=0x7feb285fb100) at /home/batsdk/software/tools/php-5.6.20/Zend/zend_variables.c:45
#3  0x00007feb21ab0fe5 in _zval_dtor (type=<value optimized out>, module_number=<value optimized out>)
    at /home/batsdk/thirdparty/php-5.6/include/php/Zend/zend_variables.h:35
#4  seaslog_clear_logger_list (type=<value optimized out>, module_number=<value optimized out>) at /home/batsdk/tools/software/SeasLog-master/seaslog.c:605
#5  zm_deactivate_seaslog (type=<value optimized out>, module_number=<value optimized out>) at /home/batsdk/tools/software/SeasLog-master/seaslog.c:274
#6  0x000000000078b894 in zend_deactivate_modules () at /home/batsdk/software/tools/php-5.6.20/Zend/zend_API.c:2449
#7  0x00000000007250f5 in php_request_shutdown (dummy=<value optimized out>) at /home/batsdk/software/tools/php-5.6.20/main/main.c:1867
#8  0x00000000008335f9 in main (argc=<value optimized out>, argv=<value optimized out>) at /home/batsdk/software/tools/php-5.6.20/sapi/fpm/fpm/fpm_main.c:2017

**Comment by Neeke** [comment]
@yyz26371945   你的QQ号多少

**Comment by yyz26371945** [comment]
340427874

**Comment by Neeke** [comment]
Fixed with 1.6.8
